### PR TITLE
fix(settings): restore paste-to-destructure for workspace secrets, cleanup hooks and design tokens

### PR DESCRIPTION
**Pull request by waleedlatif1**
## Summary
- Restores the paste-to-destructure feature in Settings → Secrets that was lost when the Credentials tab split into Secrets and Integrations
- Pasting \`KEY=VALUE\` into a workspace secret row now splits into key/value fields; pasting multiple lines creates multiple rows
- Handles \`export\` prefix, quoted values, inline comments, and base64 false positives
- Adds consistent value masking to new workspace input rows (show on focus, mask on blur) matching existing saved-row behavior
- Removes ~20 unnecessary \`useCallback\` wrappers (none had observed references)
- Fixes direct state mutation in \`handleSingleValuePaste\` (was mutating prior state object in-place)
- Moves \`e.preventDefault()\` inside the \`parsedVars.length > 0\` guard so \`KEY=\` pastes don't silently swallow input
- Replaces all hardcoded hex colors with CSS variable tokens (\`--error-muted\`, \`--success\`, etc.)
- Converts all template-literal \`className\` to \`cn()\`
- Replaces raw \`<button>\` with emcn \`Button variant='ghost'\`

## Type of Change
- [x] Bug fix

## Testing
Tested manually — pasted single \`KEY=VALUE\`, multi-line env block, plain values, and edge cases (quoted values, inline comments, export prefix)

## Checklist
- [x] Code follows project style guidelines
- [x] Self-reviewed my changes
- [ ] Tests added/updated and passing
- [x] No new warnings introduced
- [x] I confirm that I have read and agree to the terms outlined in the [Contributor License Agreement (CLA)](./CONTRIBUTING.md#contributor-license-agreement-cla)

### CODE REVIEW

File: apps/sim/app/workspace/[workspaceId]/settings/components/credentials/credentials-manager.tsx [hunk 1]
```diff
@@ -833,130 +871,107 @@ export function CredentialsManager() {
...
-      setEnvVars((prev) => {
-        const newEnvVars = [...prev]
-        newEnvVars[index][inputType] = text
-        return newEnvVars
-      })
-    },
-    []
-  )
+    handleKeyValuePaste(lines)
+  }
```
[the remark this answers is not in the document]
Comment (waleedlatif1) [reply apps/sim/app/workspace/[workspaceId]/settings/components/credentials/credentials-manager.tsx:930]: Fixed in d98bd8c. Restructured `handlePaste` to mirror `handleWorkspacePaste`: moved `e.preventDefault()` out of the top of the function and into two targeted guards — one for the single-value path (no KV pattern detected) and one inside `if (parsedVars.length > 0)` for the multi-value path. A `KEY=` paste with a valid key but empty value now falls through to native browser paste instead of being silently swallowed.

File: apps/sim/app/workspace/[workspaceId]/settings/components/credentials/credentials-manager.tsx [hunk 2]
```diff
@@ -121,6 +122,59 @@ function validateEnvVarKey(key: string): string | undefined {
...
+
+  value = value.trim()
+
+  if (
+    (value.startsWith('"') && value.endsWith('"')) ||
+    (value.startsWith("'") && value.endsWith("'")) ||
+    (value.startsWith('`') && value.endsWith('`'))
+  ) {
+    value = value.slice(1, -1)
+  }
```
[the remark this answers is not in the document]
Comment (waleedlatif1) [reply apps/sim/app/workspace/[workspaceId]/settings/components/credentials/credentials-manager.tsx:166]: Fixed in d98bd8c. Added `value.length >= 2` guard before the quote-stripping condition so a lone quote character (`KEY="`) can't satisfy both `startsWith` and `endsWith`, preventing `slice(1, -1)` from returning an empty string that then gets silently dropped by the `key && value` filter.

**Comment by waleedlatif1** [comment]
@greptile

**Comment by waleedlatif1** [comment]
@cursor review

File: apps/sim/app/workspace/[workspaceId]/settings/components/credentials/credentials-manager.tsx [hunk 3]
```diff
@@ -833,130 +872,112 @@ export function CredentialsManager() {
...
-    if (
-      (value.startsWith('"') && value.endsWith('"')) ||
-      (value.startsWith("'") && value.endsWith("'")) ||
-      (value.startsWith('`') && value.endsWith('`'))
-    ) {
-      value = value.slice(1, -1)
+    const parsedVars = parseValidEnvVars(lines)
+    if (parsedVars.length > 0) {
+      e.preventDefault()
+      handleKeyValuePaste(lines)
```
[the remark this answers is not in the document]
Comment (waleedlatif1) [reply apps/sim/app/workspace/[workspaceId]/settings/components/credentials/credentials-manager.tsx]: Fixed in 2b3be11. Removed `handleKeyValuePaste` entirely — it was a single-use wrapper that forced a second `parseValidEnvVars` call. `handlePaste` now computes `parsedVars` once, gates `e.preventDefault()` on `parsedVars.length > 0`, and applies the state update inline. One parse, one result, no wasted `generateRowId()` calls.

**Comment by waleedlatif1** [comment]
@greptile

**Comment by waleedlatif1** [comment]
@cursor review
